artichoke-backend/src/extn/core/integer/mod.rs

```diff
@@ -50,7 +50,7 @@ impl Integer {
                 &interp,
                 format!(
                     "encoding parameter of Integer#chr (given {}) not supported",
-                    encoding.inspect()
+                    String::from_utf8_lossy(&encoding.inspect()).to_string()
```

Comment:
There is an API to construct an exception from a byte vector called `NotImplementedError::new_raw`. You'd use it like:

https://github.com/artichoke/artichoke/blob/2421bfcc24a9035575319a5205f10252437cb0a1/artichoke-backend/src/extn/core/env/backend/memory.rs#L74-L78